### PR TITLE
[2.7] create crtb for restricted admin

### PR DESCRIPTION
**Pull request by KevinJoiner**
Forward Port of https://github.com/rancher/rancher/pull/41077 and https://github.com/rancher/rancher/pull/41217
This was a cherry-pick of 007e03f7d2a585cbccad72123262c834981c70ac 5c2f63366250e516de1d0e9b88a88851ba039e1b 8ec605015a8ea36c3c3b766a61b9c539c15efec7 b22a57e00af6e45983bfdd2de8c3b721fb3e9cb2 85721680e153f638bf2c189c3899928a20f7dabc  (https://github.com/rancher/rancher/pull/41077) and 15486ffb89e382d6b764e2dc76760c7b96451eb8 fd808b2922d98af65a276ea23ed5d2ba3635621f (https://github.com/rancher/rancher/pull/41217) with no conflicts.

**Notice:**
There is one extra commit ff2bb296c7b29013c5567acfdf42f0b08e096af7 that is not a cherry-pick and should be reviewed. This commit fixes a simple logic error fo checking for a notFoundError before checking if the error was nil. This commit also adds unit tests for the cluster_handler.

### CODE REVIEW

**Comment by KevinJoiner** [comment]
rebased with [release/v2.7](https://github.com/rancher/rancher/tree/release/v2.7)

**Review by MbolotSuse — APPROVED** [review]
IDK if you need to add any extra tests for this since it's in the next 2.7 public release, but the code LGTM as a straight-forward forward-port of the earlier issue.

**Review by cmurphy — COMMENTED** [review]
needs a rebase

File: pkg/controllers/managementuser/rbac/handler_base.go [hunk 1]
```diff
@@ -18,7 +18,6 @@ import (
 	nsutils "github.com/rancher/rancher/pkg/namespace"
 	pkgrbac "github.com/rancher/rancher/pkg/rbac"
 	"github.com/rancher/rancher/pkg/types/config"
-	"github.com/rancher/wrangler/pkg/relatedresource"
```
Comment (MbolotSuse) [inline pkg/controllers/managementuser/rbac/handler_base.go:21]: Can you remove this import? I see it still being used on L135 (unlike in the 2.6 branch).

**Review by MbolotSuse — COMMENTED** [review]
LGTM, will approve if/when CI passes.

**Review by MbolotSuse — APPROVED** [review]

**Review by cmurphy — APPROVED** [review]

**Pull request merged**